test/parallel/sdk.API.claimRewards.test.ts

```diff
@@ -0,0 +1,450 @@
+        defaultCurrencyValue.divn(2)
+      )
+    ),
+    Sudo.sudoAs(
+      testUser1,
+      Xyk.mintLiquidity(
+        MGA_ASSET_ID,
+        poolTokenIds[1],
+        defaultCurrencyValue.divn(2)
+      )
+    )
+  );
+
+  await waitForRewards(testUser1, liqIds[1]);
+
+  const rewardsLiqId1Before = await getRewardsInfo(
+    testUser1.keyRingPair.address,
+    liqIds[0]
+  );
+
+  const rewardsLiqId2Before = await getRewardsInfo(
+    testUser1.keyRingPair.address,
+    liqIds[1]
+  );
+
+  await claimRewardsAll(testUser1).then((result) => {
+    const eventResponse = getEventResultFromMangataTx(result);
+    expect(eventResponse.state).toEqual(ExtrinsicResult.ExtrinsicSuccess);
+  });
+
+  const rewardsLiqId1After = await getRewardsInfo(
+    testUser1.keyRingPair.address,
+    liqIds[0]
+  );
+
+  const rewardsLiqId2After = await getRewardsInfo(
+    testUser1.keyRingPair.address,
+    liqIds[1]
+  );
+
+  expect(rewardsLiqId1Before.rewardsAlreadyClaimed).bnEqual(BN_ZERO);
+  expect(rewardsLiqId1After.rewardsAlreadyClaimed).bnGt(BN_ZERO);
+  expect(rewardsLiqId2Before.rewardsAlreadyClaimed).bnEqual(BN_ZERO);
+  expect(rewardsLiqId2After.rewardsAlreadyClaimed).bnGt(BN_ZERO);
+});
+
+test("GIVEN an user has available some rewards in two pools ( one deactivated ) WHEN claims all rewards THEN the user gets the rewards for that's pools", async () => {
+  await Sudo.batchAsSudoFinalized(
+    Assets.mintToken(poolTokenIds[0], testUser1, defaultCurrencyValue),
+    Assets.mintToken(poolTokenIds[12], testUser1, defaultCurrencyValue),
+    Sudo.sudoAs(
+      testUser1,
+      Xyk.mintLiquidity(
+        MGA_ASSET_ID,
+        poolTokenIds[0],
+        defaultCurrencyValue.divn(2)
+      )
+    ),
+    Sudo.sudoAs(
+      testUser1,
+      Xyk.mintLiquidity(
+        MGA_ASSET_ID,
+        poolTokenIds[12],
+        defaultCurrencyValue.divn(2)
+      )
+    )
+  );
+
+  await waitForRewards(testUser1, liqIds[12]);
+
+  await promotePool(sudo.keyRingPair, liqIds[12], 0);
+
+  const rewardsLiqId1Before = await getRewardsInfo(
+    testUser1.keyRingPair.address,
+    liqIds[0]
+  );
+
+  const rewardsLiqId2Before = await getRewardsInfo(
+    testUser1.keyRingPair.address,
+    liqIds[12]
+  );
+
+  await claimRewardsAll(testUser1).then((result) => {
+    const eventResponse = getEventResultFromMangataTx(result);
+    expect(eventResponse.state).toEqual(ExtrinsicResult.ExtrinsicSuccess);
+  });
+
+  const rewardsLiqId1After = await getRewardsInfo(
+    testUser1.keyRingPair.address,
+    liqIds[0]
+  );
+
+  const rewardsLiqId2After = await getRewardsInfo(
+    testUser1.keyRingPair.address,
+    liqIds[12]
+  );
+
+  expect(rewardsLiqId1Before.rewardsAlreadyClaimed).bnEqual(BN_ZERO);
+  expect(rewardsLiqId1After.rewardsAlreadyClaimed).bnGt(BN_ZERO);
+  expect(rewardsLiqId2Before.rewardsAlreadyClaimed).bnEqual(BN_ZERO);
+  expect(rewardsLiqId2After.rewardsAlreadyClaimed).bnGt(BN_ZERO);
+
+  await promotePool(sudo.keyRingPair, liqIds[12], 20);
```

Comment:
do this in a after each, because of this test fail, the environment is not right for the following tests.

Reply:
I created a special pool, which we use only for this test

Reply:
Ok , deleted.